### PR TITLE
Flexible schedule

### PR DESCRIPTION
**Pull request by fsi206914**
Did some Refactor to FrontEnd code. Our moment.js in Azkaban is too old to be compatible with PrettyCron.js. Then I fix a minor issue in PrettyCron.js to let it work. Referring to Issue #693

### CODE REVIEW

**Comment by HappyRay** [comment]
@fsi206914 could you resolve the conflicts?

Could you add a link to the issue #693 in your change description?

File: azkaban-webserver/src/web/js/azkaban/view/schedule-panel.js [hunk 1]
```diff
@@ -39,40 +37,32 @@ azkaban.SchedulePanelView = Backbone.View.extend({
...
     var scheduleURL = contextURL + "/schedule"
     var scheduleData = flowExecuteDialogView.getExecutionOptionData();
 
-    console.log("Creating schedule for " + projectName + "." +
-        scheduleData.flow);
+    console.log("Creating schedule for " + projectName + "." + scheduleData.flow);
 
-    var scheduleTime = moment(timeVal, 'h/mm A').format('h,mm,A,') + timezoneVal;
-    console.log(scheduleTime);
+    var currentMomentTime = moment();
```
Comment (HappyRay) [inline azkaban-webserver/src/web/js/azkaban/view/schedule-panel.js]: Does it return the time based on the browser's time zone which may be different from the server's time zone?

File: azkaban-webserver/src/main/resources/azkaban/webapp/servlet/velocity/scheduledflowpage.vm [hunk 2]
```diff
@@ -101,15 +102,24 @@
                 <td>${sched.submitUser}</td>
                 <td>$utils.formatDateTime(${sched.firstSchedTime})</td>
                 <td>$utils.formatDateTime(${sched.nextExecTime})</td>
-                <td>$utils.formatPeriod(${sched.period})</td>
+                #if (${sched.period})
```
Comment (HappyRay) [inline azkaban-webserver/src/main/resources/azkaban/webapp/servlet/velocity/scheduledflowpage.vm]: Consider making the switch to the new UI using a config value.

File: azkaban-common/src/main/java/azkaban/scheduler/Schedule.java [hunk 3]
```diff
@@ -121,9 +151,7 @@ public String getScheduleName() {
   public String toString() {
     return projectName + "." + flowName + " (" + projectId + ")"
         + " to be run at (starting) "
-        + new DateTime(firstSchedTime).toDateTimeISO()
-        + " with recurring period of "
-        + (period == null ? "non-recurring" : createPeriodString(period));
+        + new DateTime(firstSchedTime).toDateTimeISO();
```
Comment (HappyRay) [inline azkaban-common/src/main/java/azkaban/scheduler/Schedule.java]: Add cron info?

File: azkaban-common/src/main/java/azkaban/utils/Utils.java [hunk 4]
```diff
@@ -455,4 +461,25 @@ public static long parseMemString(String strMemSize) {
...
+   * @return : org.quartz.CronExpression object.
+   */
+  public static CronExpression parseCronExpression(String cronExpression, DateTimeZone timezone) {
+    if (cronExpression != null) {
+      try {
+        CronExpression ce =  new CronExpression(cronExpression);
+        ce.setTimeZone(TimeZone.getTimeZone(timezone.getID()));
+        return ce;
+      } catch (ParseException pe) {
+        pe.printStackTrace();
```
Comment (HappyRay) [inline azkaban-common/src/main/java/azkaban/utils/Utils.java]: Write to logs?

Comment (fsi206914) [inline azkaban-common/src/main/java/azkaban/utils/Utils.java]: fixed.

File: azkaban-common/src/main/java/azkaban/scheduler/Schedule.java (hunk 3 above)
Comment (fsi206914) [inline azkaban-common/src/main/java/azkaban/scheduler/Schedule.java]: fixed.

File: azkaban-webserver/src/main/resources/azkaban/webapp/servlet/velocity/scheduledflowpage.vm (hunk 2 above)
Comment (fsi206914) [inline azkaban-webserver/src/main/resources/azkaban/webapp/servlet/velocity/scheduledflowpage.vm]: fixed.

File: azkaban-webserver/src/web/js/azkaban/view/schedule-panel.js (hunk 1 above)
Comment (fsi206914) [inline azkaban-webserver/src/web/js/azkaban/view/schedule-panel.js]: fixed

**Comment by fsi206914** [comment]
Created a new pull request to fix the comment issues refer #704

**Comment by fsi206914** [comment]
#728 fixed this pull request.

**Pull request closed without merging**